File: Telegram App/MorseReferenceTableViewController.m
//
//  MorseReferenceTableViewController.m
//  Telegram App
//
//  Created by Sandeep  Raghunandhan on 8/21/14.
//  Copyright (c) 2014 Sandeep  Raghunandhan. All rights reserved.
//
#import "MorseCode.h"
#import "MorseReferenceTableViewController.h"

@interface MorseReferenceTableViewController ()
@property MorseCode *morse;
@end

@implementation MorseReferenceTableViewController

- (id)initWithStyle:(UITableViewStyle)style
{
    self = [super initWithStyle:style];
    if (self) {
        // Custom initialization
        _morse = [[MorseCode alloc]init];
        UINavigationItem *nav = [self navigationItem];
        nav.title = @"Morse Reference";
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"reuseIdentifier"];
    
    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;
    
    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    // Return the number of sections.
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    // Return the number of rows in the section.
    NSLog(@"%lu", (unsigned long)_morse.characters.count);
    return [_morse.characters count];
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"reuseIdentifier" forIndexPath:indexPath];
    
    // Configure the cell...
    long index = indexPath.row;
    NSMutableString *cellText = [[NSMutableString alloc]init];
    [cellText appendString:_morse.characters[index]];
    [cellText appendString:@"\t"];
    [cellText appendString:_morse.morseCodes[index]];
    cell.textLabel.text = cellText;
    
    return cell;
}

-(NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section
{
    return @"Morse Code Reference";
}


@end

File: Telegram App/MorseReferenceTableViewController.swift
//
//  MorseReferenceTableViewController.swift
//  Telegram App
//
//  Created by Sandeep on 11/18/16.
//  Copyright © 2016 Sandeep. All rights reserved.
//

import UIKit

class MorseReferenceTableViewController: UITableViewController{
    
    var morse: MorseCode!;
    private override init(nibName nibNameOrNil: String!, bundle nibBundleOrNil: Bundle!) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }
    convenience init() {
        self.init(nibName: nil, bundle: nil);
        let nav = UINavigationItem();
        nav.title="Morse Reference";
        morse = MorseCode();
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder);
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "reuseIdentifier");
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: - Table view data source

    override func numberOfSections(in tableView: UITableView) -> Int {
        // #warning Incomplete implementation, return the number of sections
        return 1
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        // #warning Incomplete implementation, return the number of rows
        return (morse?.characters?.count)!;
    }

    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "reuseIdentifier", for: indexPath)
        
        // Configure the cell...
        let index = indexPath.row;
        let cellText = String(describing: morse.characters[index]) + "\t" + String(describing: morse.morseCodes[index]);
        cell.textLabel?.text = cellText;
        return cell
    }
    
    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
        return "Morse Code Reference"; 
    }


}
